lua/nvim-treesitter/install.lua

```diff
@@ -284,7 +284,11 @@ local function run_install(cache_folder, install_folder, lang, repo, with_sync,
     ) .. '" are not executable.')
     return
   end
-  local revision = configs.get_update_strategy() == "lockfile" and get_revision(lang)
+
+  local revision = repo.revision or repo.branch
+  if revision == nil then
```

Comment:
I think you want here only `repo.revision`, since `repo.branch` should follow the lockfile. Maybe a setting to just disable the lockfile for this parser would be clearer?

Reply:
For my usage `repo.branch` does not follow the lockfile. I want to use the `nvimc` branch of https://github.com/nvim-treesitter/tree-sitter-c